server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/DBSchemaIndexService.java

```diff
@@ -0,0 +1,207 @@
+                if (!Objects.equals(authenticationFacade.currentUserId(), config.getCreatorId())) {
+                    throw new NotFoundException(ResourceType.ODC_CONNECTION, "id", params.getDatasourceId());
+                }
+                if (CollectionUtils.isNotEmpty(params.getDatabaseIds())) {
+                    queryDatabaseIds.addAll(params.getDatabaseIds().stream().filter(id2Database::containsKey)
+                            .collect(Collectors.toList()));
+                } else {
+                    queryDatabaseIds.addAll(id2Database.keySet());
+                }
+            } else {
+                Set<Long> projectIds = projectService.getMemberProjectIds(authenticationFacade.currentUserId());
+                if (CollectionUtils.isNotEmpty(params.getDatabaseIds())) {
+                    queryDatabaseIds.addAll(params.getDatabaseIds().stream()
+                            .filter(e -> id2Database.containsKey(e) && id2Database.get(e).getProject() != null
+                                    && projectIds.contains(id2Database.get(e).getProject().getId()))
+                            .collect(Collectors.toList()));
+                } else {
+                    queryDatabaseIds.addAll(id2Database.values().stream()
+                            .filter(e -> e.getProject() != null && projectIds.contains(e.getProject().getId()))
+                            .map(Database::getId).collect(Collectors.toList()));
+                }
+            }
+        } else {
+            throw new IllegalArgumentException("projectId or datasourceId is required");
+        }
+        if (CollectionUtils.isEmpty(queryDatabaseIds)) {
+            return resp;
+        }
+        List<Database> databases = databaseService.listDatabasesDetailsByIds(queryDatabaseIds);
+        Map<Long, Database> id2Database =
+                databases.stream().collect(Collectors.toMap(Database::getId, e -> e, (e1, e2) -> e1));
+        if (CollectionUtils.isEmpty(params.getTypes()) || params.getTypes().contains(DBObjectType.SCHEMA)) {
+            String key = params.getSearchKey().toLowerCase();
+            resp.setDatabases(databases.stream().filter(e -> {
+                String name = e.getName().toLowerCase();
+                if (key.startsWith("%") && key.endsWith("%")) {
+                    return name.contains(key.substring(1, key.length() - 1));
+                } else if (key.startsWith("%")) {
+                    return name.endsWith(key.substring(1));
+                } else if (key.endsWith("%")) {
+                    return name.startsWith(key.substring(0, key.length() - 1));
+                } else {
+                    return e.getName().equals(key);
+                }
+            }).collect(Collectors.toList()));
+        }
+        if (CollectionUtils.isEmpty(params.getTypes()) || params.getTypes().contains(DBObjectType.COLUMN)) {
```

Comment:
sam as above, I think that we should use spec to do this nor doing this by ourselves~

Reply:
We should select from different table depends on different request, so I do these by codes. These will be also effective.